### PR TITLE
CUMULUS-2705: Add schemaspy generated doc/images

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-2705: Add schemaspy docs](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2705)

## Changes

* Updates Core doc generation to utilize https://schemaspy.org/ to automatically generate an ERD based on the migrated schema. 

It's notable and debate-able that this change *does* require the unit test stack to be running, or at least the DB container , for this to succeed.    Posting for review to get commentary on approach/etc.  

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -40,6 +40,14 @@ Cumulus entity state data is stored in a set of [DynamoDB](https://aws.amazon.co
 
 Discovering data for ingest is handled via workflow step components using Cumulus `provider` and `collection` configurations and various triggers.    Data can be ingested from AWS S3, FTP, HTTPS and more.
 
+#### Database
+
+Cumulus utililzes a user-provided PostgreSQL database backend with data replication to an elasticsearch instance utilized for API search query efficency.   For legacy reasons, Core is currently also deploying a DynamoDB datastore that writes are replicated to in parallel with the PostgreSQL database writes.   These data stores will be removed in future releases.
```
Comment (Jkovarik) [inline docs/architecture.md:45]: Open to feedback re: this, it's intended to be a true, merge-able, but not replacement for @laurenfrederick 's RDS arch doc work.

Comment (laurenfrederick) [reply docs/architecture.md:45]:
```suggestion
Cumulus utilizes a user-provided PostgreSQL database backend. For improved API search query efficiency Cumulus provides data replication to an Elasticsearch instance.   For legacy reasons, Cumulus is currently also deploying a DynamoDB datastore, and writes are replicated in parallel with the PostgreSQL database writes.   The DynamoDB replicated tables and parallel writes will be removed in future releases.
```

Comment (laurenfrederick) [reply docs/architecture.md:45]: I had to read that first sentence a few times so reworded it. Then went on a rewording spree.

**Review by laurenfrederick — APPROVED** [review]
This is awesome, nice work! I just left a wording suggestion in there.

**Pull request merged**